python/cudf_polars/cudf_polars/experimental/io.py

```diff
@@ -352,3 +353,47 @@ def _(
         return new_node, partition_info
 
     return ir, {ir: PartitionInfo(count=1)}  # pragma: no cover
+
+
+@lower_ir_node.register(Sink)
+def _(
+    ir: Sink, rec: LowerIRTransformer
+) -> tuple[IR, MutableMapping[IR, PartitionInfo]]:
```

Comment:
I don't know if https://github.com/rapidsai/cudf/pull/18964 is worth doing (it got less useful as I went further in the implementation), but this could be tightened up to return 

```suggestion
) -> tuple[Sink, MutableMapping[IR, PartitionInfo]]:
```

I think.